### DISCUSSION TITLE
ability to fetch random record

### DISCUSSION THREAD

**Discussion by jnsougata** [deta/cloud-docs#338] [Ideas] [answered]
Some times if one needs a random record from from their base they got to have something sequential inside the record otherwise it's kinda hacky to fetch everything from the base then choosing a random record from it. 
If it's implemented in the backend and there's an endpoint for it, will be a lot more efficient and faster.

**something like this:**

```python
base.fetch(random=True)
```

**Comment by abdelhai** [comment, marked as answer]
i'm not sure this is a common use case and is worth implementing. the workarounds we presented seems to do the job for now.

**Reply by jnsougata** [reply]
Let's say someone is creating a trivia app for that they have 100K questions stored in a base. Now they created to an API to fetch a random trivia question from the base. The volume of the questions may vary. So if they want to fetch a random question the randomized choice from the back-end will be more logical as it won't error for the change in volume. 

As for use cases:
1. Trivia Questions
2. Random Jokes
3. Picture carousel

**Reply by abdelhai** [reply]
the solution i provided (with the timestamps) should work just fine. we only implement big features like this when there's enough demand
